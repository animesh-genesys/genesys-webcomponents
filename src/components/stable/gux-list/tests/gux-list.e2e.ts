import { newSparkE2EPage, a11yCheck } from 'test/e2eTestUtils';

describe('gux-list', () => {
  describe('#render', () => {
    it('should render component as expected', async () => {
      const html = `
        <gux-list>
          <gux-list-item>Test1</gux-list-item>
          <gux-list-divider></gux-list-divider>
          <gux-list-item>Test2</gux-list-item>
          <gux-list-item>Test3</gux-list-item>
          <gux-list-item>Test4</gux-list-item>
        </gux-list>
      `;
      const page = await newSparkE2EPage({ html });
      const element = await page.find('gux-list');
      await a11yCheck(page);

      expect(element.outerHTML).toMatchSnapshot();
    });
  });
});
