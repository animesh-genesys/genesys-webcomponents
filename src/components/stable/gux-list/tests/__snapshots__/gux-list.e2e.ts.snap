// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`gux-list renders 1`] = `"<gux-list hydrated=\\"\\"> <gux-list-item value=\\"test1\\" text=\\"first\\" role=\\"listitem\\" hydrated=\\"\\"></gux-list-item> <gux-list-divider hydrated=\\"\\"></gux-list-divider> <gux-list-item value=\\"test2\\" text=\\"second\\" disabled=\\"\\" role=\\"listitem\\" hydrated=\\"\\"></gux-list-item> <gux-list-item value=\\"test3\\" text=\\"third\\" role=\\"listitem\\" hydrated=\\"\\"></gux-list-item> </gux-list>"`;
